doc/reference/actions.rst

```diff
@@ -284,6 +304,22 @@ The following constructions are available:
        (run ./list-tests.exe)
        (run ./exec-tests.exe))
 
+
+.. dune:action:: when
+   :param: <condition> <action>
+
+   Conditionally perform an action. ``<condition>`` is an expression in the
+   :doc:`boolean-language` and ``<action>`` is some action.
+
+   Example::
+
+     ; prints "You're using Linux!" or "You're using MacOS!"
+
+     (progn
+      (echo "You're using ")
+      (when (= %{system} linux) (echo "Linux!"))
+      (when (= %{system} macosx) (echo "MacOS!")))
```

Comment:
At first I thought this could be a simpler replacement for `case` and `cond`, but upon further inspection we lose the ability to stop at the first branch. Also another way to write this would be to use `concurrent` (on the last two args anyway). That way you wouldn't have to sequentially do all the checks.

Reply:
It's deliberately much simpler than `case` and `cond` and it's supposed to behave like the `when` keyword in many lisp languages. In practice it's going to be used to filter out opam commands. It might not be practically useful to expose this in non-pkg files, at least for now. @rgrinberg what do you think?

Reply:
>Also another way to write this would be to use `concurrent` (on the last two args anyway). That way you wouldn't have to sequentially do all the checks.

Do you just mean as an optimization?

Reply:
> Possibly this example is misleading as it makes it look like a case statement when it isn't one. @rgrinberg what do you think?

I'd rather introduce whatever is needed to keep lock files as small and as readable as possible.

Reply:
To clarify I was asking whether it makes sense for `when` to be available in `dune` files or only in lock files.

Reply:
At the moment? There's no need to make it available. Eventually, yes the plan is to completely unify dune and lock files.